HPXMLtoOpenStudio/resources/schedules.rb

```diff
@@ -1442,8 +1447,28 @@ def convert_setpoints
     end
   end
 
+  def battery_schedules
+    return if !@schedules.keys.include?(SchedulesFile::ColumnBattery)
+
+    @schedules[SchedulesFile::ColumnBatteryCharging] = Array.new(@schedules[SchedulesFile::ColumnBattery].size, 0)
+    @schedules[SchedulesFile::ColumnBatteryDischarging] = Array.new(@schedules[SchedulesFile::ColumnBattery].size, 0)
+    @schedules[SchedulesFile::ColumnBattery].each_with_index do |_ts, i|
+      if @schedules[SchedulesFile::ColumnBattery][i] > 0
+        @schedules[SchedulesFile::ColumnBatteryCharging][i] = @schedules[SchedulesFile::ColumnBattery][i]
+        @schedules[SchedulesFile::ColumnBatteryDischarging][i] = 0
+      elsif @schedules[SchedulesFile::ColumnBattery][i] < 0
+        @schedules[SchedulesFile::ColumnBatteryCharging][i] = 0
+        @schedules[SchedulesFile::ColumnBatteryDischarging][i] = -1 * @schedules[SchedulesFile::ColumnBattery][i]
+      else
+        @schedules[SchedulesFile::ColumnBatteryCharging][i] = 0
+        @schedules[SchedulesFile::ColumnBatteryDischarging][i] = 0
+      end
```

Comment:
No need to set zeroes since you've initialized the arrays to all zeroes.

Reply:
Actually, this entire method seems overly complicated for what it's trying to do... isn't it this simple?
```
@schedules[SchedulesFile::ColumnBatteryCharging] = @schedules[SchedulesFile::ColumnBattery]
@schedules[SchedulesFile::ColumnBatteryDischarging] = @schedules[SchedulesFile::ColumnBattery].map { |v| -1 * v }
```